[Feature] Unify Load Planner

xiaoyong-z opened the pull request:
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #11670

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

chaoyli commented on be/src/connector/file_connector.cpp:
```diff
@@ -107,7 +107,7 @@ Status FileDataSource::get_next(RuntimeState* state, vectorized::ChunkPtr* chunk
 
         const TQueryOptions& query_options = state->query_options();
-        if (query_options.__isset.load_job_type && query_options.load_job_type == TLoadJobType::Broker) {
+        if (query_options.__isset.load_job_type && query_options.load_job_type == TLoadJobType::BROKER) {
```
It's better to add a concise message about the feature in the commit message and related issues.

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:
```diff
@@ -694,8 +695,18 @@ public TExecPlanFragmentParams plan(TUniqueId loadId, long txnId) throws UserExc
+                }
+                InsertPlannerV2 insertPlanner = new InsertPlannerV2(id, loadId, txnId, db.getId(), (OlapTable) table, 
+                        isStrictMode(), getTimezone(), isPartialUpdate(), ConnectContext.get(), sessionVariables, 
+                        execMemLimit, execMemLimit, false, getColumnDescs(), StreamLoadTask.fromRoutineLoadJob(this));
```
Why to pass two execMemLimit in the parameter list?

xiaoyong-z replied:
To be consistent with the behaviour of broker load

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/sql/InsertPlannerV2.java:
```diff
@@ -0,0 +1,892 @@
+    public InsertPlannerV2(long loadJobId, TUniqueId loadId, long txnId, long dbId, OlapTable destTable,
+                            boolean strictMode, String timezone, long timeoutS,
+                            long startTime, boolean partialUpdate, ConnectContext context,
+                            Map<String, String> sessionVariables, long loadMemLimit, long execMemLimit, 
```
It's better to unify to one mem limit

xiaoyong-z replied:
To be consistent with the behaviour of broker load

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/sql/InsertPlannerV2.java:
```diff
@@ -0,0 +1,892 @@
+        // 1. Generate tuple descriptor
+        OlapTable olapDestTable = (OlapTable) destTable;
+        List<Column> destColumns = Lists.newArrayList();
+        if (isPrimaryKey && partialUpdate) {
```
How about the plan of PrimaryKey table?

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 309 / 613 (50.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskScheduler.java|0|11|00.00%|[226, 229, 230, 232, 255, 256, 257, 258, 259, 260, 262]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|0|21|00.00%|[689, 690, 691, 693, 695, 696, 697, 699, 700, 701, 703, 705, 706, 707, 711, 712, 713, 715, 717, 719, 734]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|4|00.00%|[1196, 1197, 1212, 1213]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaTaskInfo.java|0|7|00.00%|[159, 160, 175, 176, 195, 196, 197]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 140, 141, 142, 143, 144, 145, 146, 147, 149, 207, 208, 210]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|5|40.00%|[72, 76, 77]|
|:large_blue_circle:|com/starrocks/sql/InsertPlannerV2.java|237|470|50.43%|[172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 186, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 230, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 250, 251, 254, 255, 256, 258, 263, 264, 265, 266, 267, 268, 269, 270, 272, 274, 276, 277, 279, 281, 284, 287, 288, 289, 292, 293, 295, 298, 299, 300, 301, 302, 304, 305, 306, 309, 310, 311, 312, 314, 318, 319, 320, 323, 324, 327, 328, 329, 330, 331, 419, 421, 422, 423, 426, 427, 430, 431, 432, 434, 436, 439, 442, 446, 449, 451, 454, 455, 456, 457, 458, 470, 475, 476, 481, 482, 483, 484, 490, 491, 492, 494, 495, 496, 497, 498, 499, 500, 502, 503, 504, 505, 507, 508, 517, 518, 526, 534, 553, 554, 555, 556, 557, 558, 564, 565, 566, 567, 568, 569, 570, 572, 573, 577, 578, 579, 580, 585, 586, 587, 591, 595, 596, 597, 598, 602, 603, 604, 607, 608, 609, 610, 613, 614, 615, 618, 619, 622, 623, 624, 625, 626, 627, 628, 630, 631, 632, 635, 659, 689, 690, 691, 692, 694, 695, 696, 699, 726, 727, 728, 729, 731, 732, 734, 735, 736, 776, 777, 778, 779, 780, 781, 907, 908, 915, 916]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|37|48|77.08%|[359, 376, 377, 378, 379, 380, 381, 384, 385, 386, 393]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|23|23|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|3|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 309 / 613 (50.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskScheduler.java|0|11|00.00%|[226, 229, 230, 232, 255, 256, 257, 258, 259, 260, 262]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|0|21|00.00%|[689, 690, 691, 693, 695, 696, 697, 699, 700, 701, 703, 705, 706, 707, 711, 712, 713, 715, 717, 719, 734]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|4|00.00%|[1196, 1197, 1212, 1213]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaTaskInfo.java|0|7|00.00%|[159, 160, 175, 176, 195, 196, 197]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 140, 141, 142, 143, 144, 145, 146, 147, 149, 207, 208, 210]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|5|40.00%|[72, 76, 77]|
|:large_blue_circle:|com/starrocks/sql/InsertPlannerV2.java|237|470|50.43%|[172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 186, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 230, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 250, 251, 254, 255, 256, 258, 263, 264, 265, 266, 267, 268, 269, 270, 272, 274, 276, 277, 279, 281, 284, 287, 288, 289, 292, 293, 295, 298, 299, 300, 301, 302, 304, 305, 306, 309, 310, 311, 312, 314, 318, 319, 320, 323, 324, 327, 328, 329, 330, 331, 419, 421, 422, 423, 426, 427, 430, 431, 432, 434, 436, 439, 442, 446, 449, 451, 454, 455, 456, 457, 458, 470, 475, 476, 481, 482, 483, 484, 490, 491, 492, 494, 495, 496, 497, 498, 499, 500, 502, 503, 504, 505, 507, 508, 517, 518, 526, 534, 553, 554, 555, 556, 557, 558, 564, 565, 566, 567, 568, 569, 570, 572, 573, 577, 578, 579, 580, 585, 586, 587, 591, 595, 596, 597, 598, 602, 603, 604, 607, 608, 609, 610, 613, 614, 615, 618, 619, 622, 623, 624, 625, 626, 627, 628, 630, 631, 632, 635, 659, 689, 690, 691, 692, 694, 695, 696, 699, 726, 727, 728, 729, 731, 732, 734, 735, 736, 776, 777, 778, 779, 780, 781, 907, 908, 915, 916]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|37|48|77.08%|[359, 376, 377, 378, 379, 380, 381, 384, 385, 386, 393]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|23|23|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|3|100.00%|[]|

rickif commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 29 / 601 (04.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskScheduler.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|0|21|00.00%|[689, 690, 691, 693, 695, 696, 697, 699, 700, 701, 703, 705, 706, 707, 711, 712, 713, 715, 717, 719, 734]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|4|00.00%|[1196, 1197, 1212, 1213]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaTaskInfo.java|0|5|00.00%|[159, 160, 195, 196, 197]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|48|00.00%|[353, 355, 356, 357, 359, 361, 362, 363, 364, 365, 366, 367, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 384, 385, 386, 390, 391, 392, 393, 395, 397, 398, 399, 400, 401, 404, 405, 406, 407, 408, 409, 411, 412, 415, 416]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|0|3|00.00%|[52, 53, 56]|
|:large_blue_circle:|com/starrocks/sql/InsertPlannerV2.java|2|470|00.43%|[142, 144, 145, 154, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 186, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 205, 206, 207, 208, 209, 210, 211, 212, 213, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 230, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 249, 250, 251, 252, 253, 254, 255, 256, 258, 263, 264, 265, 266, 267, 268, 269, 270, 272, 274, 276, 277, 279, 281, 284, 287, 288, 289, 292, 293, 295, 298, 299, 300, 301, 302, 304, 305, 306, 309, 310, 311, 312, 314, 318, 319, 320, 323, 324, 327, 328, 329, 330, 331, 334, 335, 338, 339, 343, 344, 345, 346, 349, 352, 355, 356, 359, 362, 364, 365, 366, 367, 370, 371, 373, 374, 375, 377, 379, 383, 384, 385, 386, 387, 389, 390, 391, 392, 393, 394, 395, 396, 399, 402, 403, 405, 407, 408, 409, 411, 412, 419, 421, 422, 423, 426, 427, 430, 431, 432, 434, 436, 439, 442, 446, 449, 451, 454, 455, 456, 457, 458, 461, 463, 464, 465, 466, 467, 468, 469, 470, 473, 474, 475, 476, 478, 480, 481, 482, 483, 484, 486, 487, 490, 491, 492, 494, 495, 496, 497, 498, 499, 500, 502, 503, 504, 505, 507, 508, 513, 514, 515, 516, 517, 518, 524, 525, 526, 529, 530, 531, 532, 534, 536, 539, 540, 543, 544, 545, 546, 548, 549, 553, 554, 555, 556, 557, 558, 564, 565, 566, 567, 568, 569, 570, 572, 573, 577, 578, 579, 580, 585, 586, 587, 591, 595, 596, 597, 598, 602, 603, 604, 607, 608, 609, 610, 613, 614, 615, 618, 619, 622, 623, 624, 625, 626, 627, 628, 630, 631, 632, 635, 639, 641, 642, 643, 644, 645, 646, 647, 648, 649, 651, 652, 653, 655, 656, 657, 658, 659, 661, 662, 663, 667, 671, 672, 674, 675, 676, 677, 678, 679, 681, 682, 684, 685, 686, 687, 688, 689, 690, 691, 692, 694, 695, 696, 699, 701, 702, 704, 705, 706, 707, 708, 712, 718, 719, 721, 722, 724, 725, 726, 727, 728, 729, 731, 732, 734, 735, 736, 739, 740, 741, 742, 743, 745, 746, 747, 748, 749, 750, 752, 753, 754, 755, 756, 757, 760, 761, 762, 763, 764, 768, 769, 770, 771, 773, 774, 775, 776, 777, 778, 779, 780, 781, 783, 784, 787, 794, 795, 796, 797, 798, 799, 800, 801, 802, 803, 804, 805, 806, 807, 810, 821, 822, 823, 824, 825, 828, 829, 832, 834, 835, 839, 840, 843, 844, 847, 848, 849, 850, 851, 852, 853, 855, 857, 858, 859, 863, 867, 871, 875, 876, 879, 883, 887, 891, 895, 896, 899, 903, 907, 908, 911, 915, 916, 919, 923, 927]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 140, 141, 142, 143, 144, 145, 146, 147, 149, 207, 208, 210]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|5|40.00%|[75, 76, 77]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|18|23|78.26%|[1263, 1279, 1315, 1316, 1317]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

meegoo commented on fe/fe-core/src/main/java/com/starrocks/sql/InsertPlannerV2.java:
```diff
@@ -0,0 +1,929 @@
+        return null;
+    }
+
+    public void broker_load_plan() throws UserException {
```
CamelCase

xiaoyong-z committed:
[Feature] Unify Load Planner

Currently Starrocks supports importing stream load, routine load, insert into, broker load and spark load.
Except spark load, for the remaining four kinds of load will be unified to import and export syntax & architecture unified overall design afterwards, this work is the basis of unified import.

The main function of stream load, routine load, and broker load is to extract data from various sources and write them to olap tablet sink, and they all send execution plans to BE for execution. The planner generated is different from the following.
  1. Stream load, Routine load, is through StreamLoadPlanner
  2. Broker load, which is generated through LoadingTaskPlanner

In this pr, we unify the planner of broker load, stream load and routine load.

Signed-off-by: xyz <a997647204@gmail.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 331 (01.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|44|00.00%|[353, 355, 356, 357, 358, 359, 360, 361, 362, 363, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 380, 381, 382, 386, 387, 388, 389, 391, 393, 394, 395, 396, 397, 400, 401, 402, 403, 404, 405, 408, 409]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|0|267|00.00%|[66, 68, 94, 96, 97, 106, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 135, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 170, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 189, 190, 191, 192, 193, 194, 196, 201, 202, 203, 204, 205, 206, 207, 208, 210, 212, 214, 215, 217, 219, 222, 225, 226, 227, 230, 231, 233, 236, 237, 238, 239, 240, 242, 243, 244, 247, 248, 249, 250, 252, 256, 257, 258, 262, 263, 266, 267, 268, 269, 270, 273, 275, 276, 277, 280, 281, 284, 285, 286, 288, 290, 293, 296, 300, 303, 305, 308, 309, 310, 311, 312, 315, 317, 318, 319, 320, 321, 322, 323, 324, 327, 328, 329, 330, 331, 333, 335, 336, 337, 338, 339, 341, 342, 345, 346, 347, 350, 352, 353, 354, 355, 356, 357, 359, 360, 361, 362, 364, 365, 370, 371, 373, 375, 376, 377, 378, 384, 385, 386, 390, 391, 392, 393, 395, 399, 400, 403, 404, 405, 406, 408, 409, 412, 413, 414, 415, 416, 417, 424, 425, 426, 427, 428, 429, 430, 432, 433, 437, 438, 439, 440, 446, 447, 448, 452, 456, 457, 458, 459, 464, 465, 466, 469, 470, 471, 472, 475, 476, 477, 480, 481, 484, 485, 486, 487, 488, 489, 490, 492, 493, 494, 497, 501, 505, 509, 513, 514, 517, 521, 525, 529, 533, 534, 537, 541, 545, 546, 549, 553, 554, 557, 561, 565]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 139, 140, 141, 142, 143, 144, 145, 146, 148, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 331 (01.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|44|00.00%|[353, 355, 356, 357, 358, 359, 360, 361, 362, 363, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 380, 381, 382, 386, 387, 388, 389, 391, 393, 394, 395, 396, 397, 400, 401, 402, 403, 404, 405, 408, 409]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|0|267|00.00%|[66, 68, 94, 96, 97, 106, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 135, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 170, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 189, 190, 191, 192, 193, 194, 196, 201, 202, 203, 204, 205, 206, 207, 208, 210, 212, 214, 215, 217, 219, 222, 225, 226, 227, 230, 231, 233, 236, 237, 238, 239, 240, 242, 243, 244, 247, 248, 249, 250, 252, 256, 257, 258, 262, 263, 266, 267, 268, 269, 270, 273, 275, 276, 277, 280, 281, 284, 285, 286, 288, 290, 293, 296, 300, 303, 305, 308, 309, 310, 311, 312, 315, 317, 318, 319, 320, 321, 322, 323, 324, 327, 328, 329, 330, 331, 333, 335, 336, 337, 338, 339, 341, 342, 345, 346, 347, 350, 352, 353, 354, 355, 356, 357, 359, 360, 361, 362, 364, 365, 370, 371, 373, 375, 376, 377, 378, 384, 385, 386, 390, 391, 392, 393, 395, 399, 400, 403, 404, 405, 406, 408, 409, 412, 413, 414, 415, 416, 417, 424, 425, 426, 427, 428, 429, 430, 432, 433, 437, 438, 439, 440, 446, 447, 448, 452, 456, 457, 458, 459, 464, 465, 466, 469, 470, 471, 472, 475, 476, 477, 480, 481, 484, 485, 486, 487, 488, 489, 490, 492, 493, 494, 497, 501, 505, 509, 513, 514, 517, 521, 525, 529, 533, 534, 537, 541, 545, 546, 549, 553, 554, 557, 561, 565]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 139, 140, 141, 142, 143, 144, 145, 146, 148, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|

meegoo commented on fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadLoadingTask.java:
```diff
@@ -74,6 +76,8 @@ public class LoadLoadingTask extends LoadTask {
     private LoadingTaskPlanner planner;
     private ConnectContext context;
 
+    private LoadPlanner insertPlanner;
```
loadPlanner?

meegoo commented on fe/fe-core/src/main/java/com/starrocks/sql/LoadPlanner.java:
```diff
@@ -0,0 +1,567 @@
+        List<Long> partitionIds = getAllPartitionIds();
+
+        // For stream/routine load, we only need one fragment, ScanNode -> DataSink.
+        // OlapTableSink can dispatch data to corresponding node.
```
stream/routine load also need shuffle in some case

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 330 (01.52%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|44|00.00%|[353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 379, 380, 381, 385, 386, 387, 388, 390, 392, 393, 394, 395, 396, 399, 400, 401, 402, 403, 404, 407, 408]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|0|267|00.00%|[66, 68, 94, 96, 97, 106, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 135, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 170, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 189, 190, 191, 192, 193, 194, 196, 201, 202, 203, 204, 205, 206, 207, 208, 210, 212, 214, 215, 217, 219, 222, 225, 226, 227, 230, 231, 233, 236, 237, 238, 239, 240, 242, 243, 244, 247, 248, 249, 250, 252, 256, 257, 258, 262, 263, 266, 267, 268, 269, 270, 273, 275, 276, 277, 280, 281, 284, 285, 286, 288, 290, 293, 296, 301, 303, 305, 308, 309, 310, 311, 312, 315, 317, 318, 319, 320, 321, 322, 323, 324, 327, 328, 329, 330, 331, 333, 335, 336, 337, 338, 339, 341, 342, 345, 346, 347, 350, 352, 353, 354, 355, 356, 357, 359, 360, 361, 362, 364, 365, 370, 371, 373, 375, 376, 377, 378, 384, 385, 386, 390, 391, 392, 393, 395, 399, 400, 403, 404, 405, 406, 408, 409, 412, 413, 414, 415, 416, 417, 424, 425, 426, 427, 428, 429, 430, 432, 433, 437, 438, 439, 440, 446, 447, 448, 452, 456, 457, 458, 459, 464, 465, 466, 469, 470, 471, 472, 475, 476, 477, 480, 481, 484, 485, 486, 487, 488, 489, 490, 492, 493, 494, 497, 501, 505, 509, 513, 514, 517, 521, 525, 529, 533, 534, 537, 541, 545, 546, 549, 553, 554, 557, 561, 565]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|3|17|17.65%|[113, 116, 139, 140, 141, 142, 143, 144, 145, 146, 148, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/load/EtlJobType.java|2|2|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_admit_test
